lxd/storage/utils.go

```diff
@@ -969,12 +969,25 @@ func volumeIsUsedByDevice(vol api.StorageVolume, inst *db.InstanceArgs, dev map[
 		}
 	}
 
+	var volName string
+	var snapName string
+
+	if shared.IsSnapshot(vol.Name) {
+		parts := strings.SplitN(vol.Name, shared.SnapshotDelimiter, 2)
+		volName, snapName = parts[0], parts[1]
+	} else if dev["source.snapshot"] != "" {
+		// vol is not a snapshot but dev refers to one
+		return false, nil
```

Comment:
shouldnt this be an error?

Reply:
`volumeIsUsedByDevice` is just determining if `dev` describes `vol`; if `vol` isn't a snapshot but `dev` describes one, then it was correct use of the function, just the two don't match. For example, the function should return false if it gets a storage volume `virtual-machine/v1` but this disk device:
```
  v1-snap0:
    type: disk
    pool: default
    source: v1
    source.type: virtual-machine
    source.snapshot: snap0
```